cypress/integration/Popup.spec.ts

```diff
@@ -0,0 +1,70 @@
+import * as h from '../helpers';
+
+function getPopup() {
+  return cy.findByRole('dialog');
+}
+
+function getPopupTargetButton() {
+  return cy.contains('Delete Item');
+}
+
+function getCloseButton() {
+  return getPopup().find('[data-close]');
+}
```

Comment:
this function and the one above are similar to the modal helper functions, should I use those instead?

Reply:
Hmm. If I were to share some helpers I might do the opposite since `Popup` is what's common between `Popup` and `Modal`. But that's an implementation detail of the Modal. Modal uses shared helpers - those helper functions are meant to be usable eventually outside of this repository. Non-sharable helper functions should remain inside the spec file IMO.

These helper functions are different than the modal ones. They don't map as closely to the Cypress command log - these are more like "macro helpers" than just call to other commands and don't have their own logging. This isn't a huge problem on simpler tests, but can really aid in debugging larger tests.

Reply:
ah gotcha, thanks for explaining that!